frontend/src/app/main/data/event.cljs

```diff
@@ -183,10 +250,118 @@
 
     (rx/of nil)))
 
+
+(defn performance-observer-event-stream
+  []
+  (if (and (exists? js/globalThis)
+           (exists? (.-PerformanceObserver js/globalThis)))
+    (rx/create
+     (fn [subs]
+       (let [observer
+             (js/PerformanceObserver.
+              (fn [list]
+                (doseq [entry (.getEntries list)]
+                  (when (and (= "event" (.-entryType entry))
+                             (> (.-duration entry) min-browser-event-time))
+                    (rx/push!
+                     subs
+                     {::event :observer-event
+                      :duration (.-duration entry)
+                      :event-name (.-name entry)})))))]
+         (.observe observer #js {:entryTypes #js ["event"]})
+         (fn []
+           (.disconnect observer)))))
+    (rx/empty)))
+
+(defn performance-observer-longtask-stream
+  []
+  (if (and (exists? js/globalThis)
+           (exists? (.-PerformanceObserver js/globalThis)))
+    (rx/create
+     (fn [subs]
+       (let [observer
+             (js/PerformanceObserver.
+              (fn [list]
+                (doseq [entry (.getEntries list)]
+                  (when (and (= "longtask" (.-entryType entry))
+                             (> (.-duration entry) min-longtask-time))
+                    (rx/push! subs
+                              {::event :observer-longtask
+                               :duration (.-duration entry)})))))]
+         (.observe observer #js {:entryTypes #js ["longtask"]})
+         (fn []
+           (.disconnect observer)))))
+    (rx/empty)))
+
+(defn store-performace-info
+  []
+  (letfn [(micro-benchmark [state]
+            (let [start (js/performance.now)]
+              (doseq [i (range 0 1e6)]
+                (* (math/sin i) (math/sqrt i)))
+              (let [end (js/performance.now)]
+                (assoc-in state [:performance-info :bench-result] (- end start)))))
```

Comment:
I suggest replace the usage of assoc-in that requires building a vector for traveral to something like this

```clojure
(update state :performance-info assoc :bench-result (- end start))
```